sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -549,6 +574,31 @@ def create_job(
                     + "Number of Workers must be set."
                 )
 
+        # If labels were provided, apply them to job metadata
+        if labels is not None:
+            # Ensure metadata exists
+            if not hasattr(job, "metadata") or job.metadata is None:
+                job.metadata = models.V1ObjectMeta(
+                    name=getattr(job.metadata, "name", name),
+                    namespace=getattr(job.metadata, "namespace", namespace),
+                )
```

Comment:
We could reduce some duplication as such: 
```suggestion
      if annotations is not None or labels is not None:
            if not hasattr(job, "metadata") or job.metadata is None:
                job.metadata = models.V1ObjectMeta(
                    name=getattr(job.metadata, "name", name),
                    namespace=getattr(job.metadata, "namespace", namespace),
                )
            if labels is not None:
            ......
            if annotations is not None:
            ......
```

No strong opinion on the above, but I did notice that in the annotations block you have:
```
                    name=getattr(
                        job.metadata,
                        "name",
                        name,
                        namespace=getattr(job.metadata, "namespace", namespace),
                    )
```
While on the labels block, you have:
```
                   name=getattr(job.metadata, "name", name),
                   namespace=getattr(job.metadata, "namespace", namespace),
```
Is there a reason for this diff?

Reply:
@ChristianZaccaria Ohh great catch 👀 
Actually i'm not sure either how this happened.. while adding commit I saw some conflicts during pre-commit.. I think at that moment the formatting changed wrt missing bracket or something 🤔 
And I totally agree with avoiding redunduncy as you truly mentioned above...
I have re-adjusted code accordingly, please check!